### ISSUE TITLE
Publish gives: "nothing to commit, working directory clean"

### ISSUE THREAD

**Issue by nutgaard** [lerna/lerna#883]
This command `lerna publish --exact --cd-version patch --yes -m \"publish [ci skip]\"` fails because of this error: 
```
lerna ERR! execute  Error: On branch master
lerna ERR! execute nothing to commit, working directory clean
``` 

## Expected Behavior
After #807 it states that `lerna publish`should be a no-op if there wasn't any changes. 

## Current Behavior
lerna tries to commit while working directory is empty, however `GitUtilities.commit` fails when the working directory is clean [code](https://github.com/lerna/lerna/blob/master/src/commands/PublishCommand.js#L478).

## Possible Solution
If the working directory is clean at this point (e.g `this.updates.length === 0`) there is no need to commit. This could be a simple check in `gitCommitAndTagVersionForUpdates` before calling `commit`

## Steps to Reproduce (for bugs)
While clean working directory
1. lerna publish

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.0.0-rc.5",
  "packages": [
    "packages/node_modules/*"
  ],
  "version": "independent",
  "commands": {
    "bootstrap": {
      "ignore": "*"
    }
  }
}

```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
109 silly commit publish [ci skip]
110 error execute Error: On branch master
111 error execute nothing to commit, working directory clean
```
</p></details>

## Context
Build and publishing on every commit on master-branch using CI.

## Your Environment
CircleCI, and reproduced locally (OSX)

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.0.0-rc.5 |
| `npm --version`  | 3.10.8 |
| `node --version` | v7.0.0 |

**Comment by dlebedynskyi** [comment]
any progress on this issue?

**Comment by nutgaard** [comment]
Just a work around;
```
(lerna updated && lerna publish --exact --cd-version patch --yes -m \"publish [ci skip]\") || echo \"No change\"
```

**Comment by evocateur** [comment]
This was fixed in #982, released in [v2.1.0](https://github.com/lerna/lerna/releases/tag/v2.1.0)
